src/Microsoft.TestPlatform.CrossPlatEngine/TestEngine.cs

```diff
@@ -607,23 +623,49 @@ private static void ThrowExceptionIfTestHostManagerIsNull(ITestRuntimeProvider t
         }
     }
 
-    private static void ThrowExceptionIfAnyTestHostManagerIsNullOrNoneAreFound(List<TestRuntimeProviderInfo> testRuntimeProviders)
+    private static void WarnAboutNotFoundRuntimeProvidersOrThrowWhenNoneAreFound(List<TestRuntimeProviderInfo> testRuntimeProviders, IWarningLogger warningLogger)
```

Comment:
Reconsidered. This is more of a glorified null check. It should fail, because at that point we always expect to have a provider because it is used only in places where we are about to start a testhost and discover / run the source.